tools/tasks/task.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Base class for all tasks.
+ */
+export abstract class Task {
```

Comment:
Yes, I considered this idea [here](https://github.com/mgechev/angular-seed/issues/851#issuecomment-251251654). It might be a good idea but I'd prefer to keep the API minimalistic for now. Maybe in future will turn out that the most important precondition is not related to file type.